students/rfelts/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+# Russell Felts
+# Assignment 9 - Donor Classes
+
+
+import operator
+import datetime
+
+
+class Donor(object):
+    """ Handles all the information about a single donor: attributes, properties, and methods to provide access to the
+    donor-specific information that is needed.  """
+
+    def __init__(self, name, donations=None):
+        """ Initialize the Donor class """
+
+        self.name = name
+        assert type(donations) is list, "\nThe donation value must be entered as a list.\n"
+        self.donations = donations
+
+    def add_donation(self, donation_amount):
+        """ Append the donation to the donations list """
+        float(donation_amount)
+        self.donations.append(donation_amount)
+
+    @property
+    def donations_total(self):
+        """ Sums the values in the donation list """
+        return sum(self.donations)
+
+    @property
+    def compose_message(self):
+        """ Compose a thank you message listing the current/previous donation and the donor's donation total
+        :return: a String containing the thank you message for the donor """
+
+        return (f'\nTo: {self.name}\nSubject: Thank you.\n\n{self.name} thank you for your previous generous donation'
+                f' of {self.donations[-1]:<,.2f}.\nYour total donations to date are now: {self.donations_total:<,.2f}.')
+
+    def write_letter(self):
+        """ Write a file to disk with the donation message """
+
+        now = datetime.datetime.now()
+
+        with open(self.name + '_' + str(now.month) + '_' + str(now.year) + '.txt', 'w') as f:
+            f.write(self.compose_message)
+
+
+class DonorCollection:
+
+    def __init__(self):
+        """ Create an initial list of donor objects - potentially convert to be read from a file """
+
+        self.donor_list = [Donor("Lionel Messi", [100]), Donor("Cristiano Ronaldo", [5000, 25, 9450]),
+                           Donor("Gianluigi Buffon", [1000000, 2500.50]), Donor("Neymar", [25.25, 30, 99.99, 250]),
+                           Donor("Paolo Maldini", [9500, 6789.95, 250, 7500])]
+
+    @property
+    def list_donors(self):
+        """ Return a string containing the current donor names
+        :return: A string of the current donor names """
+
+        names = "\n"
+        for item in self.donor_list:
+            names += f'{item.name}\n'
+
+        return names
```

Comment:
you can use comprehension here:
`return '\n'.join([donor.name for donor in self.donor_list])`